src/ORM/Connect/MySQLiConnector.php

```diff
@@ -408,21 +408,33 @@ public function getLastErrorCode(): int
      */
     private function throwRelevantError(string $message, int $code, int $errorLevel, ?string $sql, array $parameters): void
     {
-        if ($errorLevel === E_USER_ERROR && ($code === 1062 || $code === 1586)) {
-            // error 1062 is for a duplicate entry
-            // see https://dev.mysql.com/doc/mysql-errors/8.4/en/server-error-reference.html#error_er_dup_entry
-            // error 1586 is ALSO for a duplicate entry and uses the same error message
-            // see https://dev.mysql.com/doc/mysql-errors/8.4/en/server-error-reference.html#error_er_dup_entry_with_key_name
-            preg_match('/Duplicate entry \'(?P<val>[^\']+)\' for key \'?(?P<key>[^\']+)\'?/', $message, $matches);
-            // MySQL includes the table name in the key, but MariaDB doesn't.
-            $key = $matches['key'];
-            if (str_contains($key ?? '', '.')) {
-                $parts = explode('.', $key);
-                $key = array_pop($parts);
+        if ($errorLevel === E_USER_ERROR) {
+            switch ($code) {
+                // error 1062 is for a duplicate entry
+                // see https://dev.mysql.com/doc/mysql-errors/8.4/en/server-error-reference.html#error_er_dup_entry
+                case 1062:
+                // error 1586 is ALSO for a duplicate entry and uses the same error message
+                // see https://dev.mysql.com/doc/mysql-errors/8.4/en/server-error-reference.html#error_er_dup_entry_with_key_name
+                case 1586:
```

Comment:
Swapping from `if` to `switch/case` isn't strictly necessary here but it makes it a little easier to read now that we're adding a second condition.